plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -247,3 +241,25 @@ func logErrorToSpan(span opentracing.Span, err error) {
 	ottag.Error.Set(span, true)
 	span.LogFields(otlog.Error(err))
 }
+
+func getHTTPRoundTripper(c *config.Configuration, logger *zap.Logger) (http.RoundTripper, error) {
+	if !c.TLS.Enabled {
+		return nil, nil
```

Comment:
that means you never want to support non-TLS connections?

Reply:
It means to use the `DefaultRoundTripper` supplied by Prometheus client (the client will do this on our behalf), but you make a good point that we lose our `ConnectTimeout` configuration with this approach. I've written a test to fail this code and fixed the implementation.